tests/e2e/monitoring/test_when_one_of_the_prometheus_node_down.py

```diff
@@ -0,0 +1,182 @@
+
+
+def setup(self):
+    """
+    Create project, pvc and an app pod
+    """
+
+    # Initializing
+    self.namespace_list = []
+    self.pvc_objs = []
+    self.pod_objs = []
+
+    assert create_multiple_project_and_pvc_and_check_metrics_are_collected(self)
+
+
+def teardown(self):
+    """
+    Delete app pods and PVCs
+    Delete project
+    """
+    # Delete created app pods and PVCs
+    assert pod.delete_pods(self.pod_objs)
+    assert pvc.delete_pvcs(self.pvc_objs)
+
+    # Switch to default project
+    ret = ocp.switch_to_default_rook_cluster_project()
+    assert ret, 'Failed to switch to default rook cluster project'
+
+    # Delete projects created
+    for prj in self.namespace_list:
+        prj_obj = ocp.OCP(kind='Project', namespace=prj)
+        prj_obj.delete(resource_name=prj)
+
+
+def create_multiple_project_and_pvc_and_check_metrics_are_collected(self):
+    """
+    Creates projects, pvcs and app pods
+    """
+    for i in range(5):
+        # Create new project
+        self.namespace = create_unique_resource_name('test', 'namespace')
+        self.project_obj = ocp.OCP(kind='Project', namespace=self.namespace)
+        assert self.project_obj.new_project(self.namespace), (
+            f'Failed to create new project {self.namespace}'
+        )
+        # Create PVCs
+        self.pvc_obj = create_pvc(
+            sc_name=self.sc_obj.name, namespace=self.namespace
+        )
+
+        # Create pod
+        self.pod_obj = create_pod(
+            interface_type=constants.CEPHBLOCKPOOL,
+            pvc_name=self.pvc_obj.name, namespace=self.namespace
+        )
+
+        self.namespace_list.append(self.namespace)
+        self.pvc_objs.append(self.pvc_obj)
+        self.pod_objs.append(self.pod_obj)
+
+    # Check for the created pvc metrics is collected
+    for pvc_obj in self.pvc_objs:
+        assert collected_metrics_for_created_pvc(pvc_obj.name), (
+            f"On prometheus pod for created pvc {pvc_obj.name} related data is not collected"
+        )
+    return True
+
+
+def get_the_collected_metrics_for_pvcs_when_node_down():
+    """
+    Returns false if the metric/data are present
+    """
+    pvcs_data = get_kube_pod_spec_volumes_persistentvolumeclaims_info_metric()
+    pvcs_list = pvcs_data['data']['result']
+    if not pvcs_list:
+        logger.info("When one of the node down where prometheus"
+                    " hosted shouldn't be able to get the data/metrics")
+        return True
+    return False
+
+
+@pytest.mark.usefixtures(
+    create_rbd_secret.__name__,
+    create_ceph_block_pool.__name__,
+    create_rbd_storageclass.__name__,
+    test_fixture.__name__
+)
+@pytest.mark.polarion_id("OCS-606")
+class TestWhenOneOfThePrometheusNodeDown(E2ETest):
+    """
+    When the nodes are down, there should not be any functional impact
+    on monitoring pods. All the data/metrics should be collected correctly.
+    """
+    @tier4
+    def test_when_one_of_the_prometheus_node_down(self):
```

Comment:
consider changing the name to 'test_monitoring_prometheus_node_down'. All monitoring tests can start with 'test_monitoring' followed by the test. i.e., 'prometheus node down' so test names are consistent and readable when we have a lot of them in the future.

Reply:
done